### PR TITLE
Add tree sequence equals

### LINKED ISSUE #1011: Expose new tc.equals() function to tree sequences too.
It might be useful to be able to do ts1.equals(ts2, ignore_provenance=True). Presumably this is trivial to port to tree sequences:

```
def equals(self, ts, **kwargs):
    return self.tables.equals(ts.tables, **kwargs)
```

(assuming https://github.com/tskit-dev/tskit/issues/896#issuecomment-729995432 is done)

### PR DESCRIPTION
**Pull request by benjeffery**
## Description

Adds equality methods to TreeSequence

Fixes #1011
# PR Checklist:

- [X] Tests that fully cover new/changed functionality.
- [X] Documentation including tutorial content if appropriate.
- [X] Changelogs, if there are API changes.

### CODE REVIEW

**Comment by AdminBot-tskit** [comment]
📖 Docs for this PR can be previewed [here](https://tskit-dev.github.io/tskit-docs/PR-1020/)

File: python/tests/test_highlevel.py [hunk 1]
```diff
@@ -1478,6 +1478,94 @@ def test_nbytes(self, tmp_path, ts_fixture):
         )
         assert nbytes == ts_fixture.nbytes
 
+    def test_equals(self, ts_fixture):
```
Comment (jeromekelleher) [inline python/tests/test_highlevel.py:1481]: Can't we just not have `ts_fixture`` as a parameter then, ie.`` def test_equals(self)`` ?

File: python/tskit/trees.py [hunk 2]
```diff
@@ -3014,6 +3014,56 @@ def __getstate__(self):
...
+        *,
+        ignore_metadata=False,
+        ignore_ts_metadata=False,
+        ignore_provenance=False,
+        ignore_timestamps=False,
+    ):
+        """
+        Returns True if  `self` and `other` are equal. By default, two tree sequences
+        are considered equal if their underlying tables'
+
```
Comment (jeromekelleher) [inline python/tskit/trees.py:3032]: We could reduce a bit of duplication and say that ``equals`` is defined by the ``equals`` of the underlying table collections and link to that documentation.

**Review by jeromekelleher — APPROVED** [review]
LGTM, some minor suggestions.

File: python/tests/test_highlevel.py (hunk 1 above)
Comment (benjeffery) [reply python/tests/test_highlevel.py]: Opps - left in from when I tried it out.

File: python/tskit/trees.py (hunk 2 above)
Comment (benjeffery) [reply python/tskit/trees.py]: Done.

**Comment by benjeffery** [comment]
@jeromekelleher fixed up, also used the function in tests.

**Pull request merged**